src/utils/web3React.ts

```diff
@@ -36,7 +36,9 @@ export const getLibrary = (provider): ethers.providers.Web3Provider => {
  * @see https://docs.binance.org/smart-chain/wallet/wallet_api.html#binancechainbnbsignaddress-string-message-string-promisepublickey-string-signature-string
  */
 export const signMessage = async (provider: any, account: string, message: string): Promise<string> => {
-  if (window.BinanceChain) {
+  const isMetaMaskInScope = provider.connection?.url === 'metamask'
+
+  if (window.BinanceChain && !isMetaMaskInScope) {
```

Comment:
Not sure if there's any way to know if BinanceChain is authed

Reply:
Checking